src/main/java/org/opensearch/security/action/tenancy/MultiTenancyConfigurationsUpdateTransportAction.java

```diff
@@ -0,0 +1,146 @@
+import org.opensearch.security.securityconf.impl.CType;
+import org.opensearch.security.securityconf.impl.SecurityDynamicConfiguration;
+import org.opensearch.security.securityconf.impl.v7.ConfigV7;
+import org.opensearch.security.support.ConfigConstants;
+import org.opensearch.tasks.Task;
+import org.opensearch.threadpool.ThreadPool;
+import org.opensearch.transport.TransportService;
+
+public class MultiTenancyConfigurationsUpdateTransportAction extends HandledTransportAction<MultiTenancyConfigurationsUpdateRequest, MultiTenancyConfigurationsRetrieveResponse> {
+
+    private static final Logger log = LogManager.getLogger(org.opensearch.security.action.tenancy.MultiTenancyConfigurationsUpdateTransportAction.class);
+
+    private final String securityIndex;
+    private final ConfigurationRepository config;
+    private final Client client;
+    private final ThreadPool pool;
+
+    @Inject
+    public MultiTenancyConfigurationsUpdateTransportAction(final Settings settings,
+                                             final TransportService transportService,
+                                             final ActionFilters actionFilters,
+                                             final ConfigurationRepository config,
+                                             final ThreadPool pool,
+                                             final Client client) {
+        super(MultiTenancyConfigurationsUpdateAction.NAME, transportService, actionFilters, MultiTenancyConfigurationsUpdateRequest::new);
+
+        this.securityIndex = settings.get(ConfigConstants.SECURITY_CONFIG_INDEX_NAME, ConfigConstants.OPENDISTRO_SECURITY_DEFAULT_CONFIG_INDEX);
+
+        this.config = config;
+        this.client = client;
+        this.pool = pool;
+    }
+
+    /** Load the configuration from the security index and return a copy */
+    protected final SecurityDynamicConfiguration<?> load() {
+        return config.getConfigurationsFromIndex(Collections.singleton(CType.CONFIG), false).get(CType.CONFIG).deepClone();
+    }
+
+    public Set<String> getAllConfiguredTenantNames() {
+
+        return this.config.getConfiguration(CType.TENANTS).getCEntries().keySet();
+    }
+
+    protected void validate(ConfigV7 updatedConfig) throws RuntimeException {
+        if(!updatedConfig.dynamic.kibana.private_tenant_enabled && (updatedConfig.dynamic.kibana.default_tenant).equals(ConfigConstants.MULTITENANCY_PRIVATE_TENANT_NAME) ) {
+            throw new IllegalArgumentException("Private tenant can not be disabled if it is the default tenant.");
+        }
+
+        if( (updatedConfig.dynamic.kibana.default_tenant).equals(ConfigConstants.MULTITENANCY_GLOBAL_TENANT_DEFAULT_NAME) ||
+                (updatedConfig.dynamic.kibana.default_tenant).equals(ConfigConstants.MULTITENANCY_GLOBAL_TENANT_NAME) ||
+                (updatedConfig.dynamic.kibana.default_tenant).equals(ConfigConstants.MULTITENANCY_PRIVATE_TENANT_NAME) ) {
```

Comment:
This is hard to parse and isn't clear at a glance.  Instead create a collection called something like acceptableDefaultTenants that contains those 3 names from the constants, then check if `updatedConfig.dynamic.kibana.default_tenant` is in that collection and return